python/GangaDirac/Lib/Credentials/DiracProxy.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import
+
+import os
+import subprocess
+from datetime import datetime, timedelta
+
+import Ganga.Utility.logging
+from Ganga.Core import GangaValueError
+from Ganga.GPIDev.Schema import SimpleItem
+
+from Ganga.GPIDev.Credentials2.ICredentialInfo import cache
+from Ganga.GPIDev.Credentials2.ICredentialRequirement import ICredentialRequirement
+from Ganga.GPIDev.Credentials2.exceptions import CredentialRenewalError
+from Ganga.GPIDev.Credentials2.VomsProxy import VomsProxyInfo
+
+from GangaDirac.Lib.Utilities.DiracUtilities import getDiracEnv
+
+logger = Ganga.Utility.logging.getLogger()
+
+
+class DiracProxyInfo(VomsProxyInfo):
+    """
+    A wrapper around a DIRAC proxy file
+    """
+
+    def __init__(self, requirements, check_file=False, create=False):
+        self._shell = None
+
+        super(DiracProxyInfo, self).__init__(requirements, check_file, create)
+
+    def create(self):
+        """
+        Creates the grid proxy.
+
+        Raises:
+            CredentialRenewalError: If the renewal process returns a non-zero value
+        """
+        group_command = ''
+        logger.debug('require ' + self.initial_requirements.group)
+        if self.initial_requirements.group:
+            group_command = '--group %s --VOMS' % self.initial_requirements.group
+        command = 'dirac-proxy-init --out %s %s' % (self.location, group_command)
```

Comment:
I think we may need to add the `-t` option here to be strict about our requirement to generate a correct proxy and to avoid problems when DIRAC accepts a bad one.
(unless this comes in through the group attribute above in which case feel free to ignore me :)